### DISCUSSION TITLE
Is WebAssembly accessible in V8 or node.js backends?

### DISCUSSION THREAD

**Discussion by ostryhub** [Tencent/puerts#1957] [Q&A]
Hi All.

I am trying to execute a js script using puerts V8 or node.js backends.
The js script calls WebAssembly API to load and instantiate the wasm binary module asynchronously. 
I had not problem getting this script to execute using ClearScript lib with V8.
When trying to do the same using PuerTS I am getting the 
"System.Exception: chunk:4: ReferenceError: WebAssembly is not defined" error.

Is WebAssembly accessible in V8 or node.js backends? by default or does it need to be enabled within custom puerts build ?
Any help is greatly aperciated!

**Comment by chexiongsheng** [comment]
Whether WebAssembly is accessible depends on the operating system. For example, it is not available on ios. Since it is not available on all operating systems, it is not recommended to use WebAssembly if you want to be cross-platform.

**Comment by ostryhub** [comment]
@chexiongsheng Hi and thank you for the answer.

> Whether WebAssembly is accessible depends on the operating system

I've tried loading the wasm script with puerts running in the unity editor so WebAssembly is actually not accessible there.
(macos on apple silicon).


> For example, it is not available on ios

I am aware of iOS clients JIT limitations, and I am not interested in iOS at all


> it is not recommended to use WebAssembly if you want to be cross-platform

I am really focusing on STANDALONE and ANDROID platforms here so I am definitely not expecting a fully cross-platform support for WebAssebly.


Knowing all the above constraints: is there a way to enable WebAssembly access through puerts for standalone and android platforms ?

Thanks!

**Comment by chexiongsheng** [comment]
Debug version of mac plugins disable the WebAssembly. You can use a release version.

**Comment by ostryhub** [comment]
@chexiongsheng Hi again,
can you elaborate on how to obtain the non debug build of the puerts plugins?

I can't find anything about this (debug vs release builds of the PuerTS) mentioned in the docs and faq.

I installed the PuerTS via Unity's UPM from local clone of ```https://github.com/chexiongsheng/puerts_unity_demo.git```
The puerts_unity_demo/package/Plugins/macOS folder has single set of dylib and bundle files with no distinctiion if they're debug or release.

Please explain how to "use a release version" ?

**Comment by chexiongsheng** [comment]
download here: https://github.com/Tencent/puerts/releases
or build it yourself according to the guide: https://puerts.github.io/en/docs/puerts/unity/other/building
